enterprise/frontend/src/embedding/data-picker/DataSelector/DataSelector.jsx

```diff
@@ -366,16 +369,29 @@ export class UnconnectedDataSelector extends Component {
     } else if (
       // Schema id is explicitly set when going through the New > Question/Model flow,
       // whereas we have to obtain it from the state when opening a saved question.
-      (this.state.selectedSchemaId || this.state.selectedSchema?.id) &&
-      steps.includes(TABLE_STEP)
+      this.state.selectedSchemaId ||
+      this.state.selectedSchema?.id
     ) {
       await this.switchToStep(TABLE_STEP);
-    } else if (this.state.selectedDatabaseId && steps.includes(SCHEMA_STEP)) {
-      await this.switchToStep(SCHEMA_STEP);
-    } else if (steps[0] === DATA_BUCKET_STEP && !this.hasUsableModels()) {
-      await this.switchToStep(steps[1]);
+    } else if (this.isJoinStep()) {
+      const isQuerySourceModel = this.props.isQuerySourceModel;
+
+      if (isQuerySourceModel) {
+        await this.switchToStep(
+          DATABASE_STEP,
+          {
+            selectedDataBucketId: "models",
+          },
+          false,
+        );
+      } else {
+        // query source is a table
+        await this.switchToStep(SCHEMA_STEP);
```

Comment:
This is the same as before. We don't need to do anything special here since the schema step will automatically select the selected database `props.selectedDatabaseId`.